Main Menu Development

karastoyanov opened the pull request:
This PR should track the progress of merging `main_menu` branch into `development`.

⚠️ The pull-request should be merged only after the Main Menu window is completly done.

karastoyanov committed:
Merge pull request #2 from Stake-And-Rope/development

gitignore file updated

karastoyanov committed:
Merge pull request #9 from Stake-And-Rope/development

Release market-app v1.0.0-alpha

karastoyanov committed:
Basic Main Menu Windows

karastoyanov committed:
Main Menu updated

karastoyanov committed:
Main Menu rework

karastoyanov committed:
Main Menu Reworked

rayapetkova committed:
Update left_buttons_groupbox

Made a left buttons group box and added a layout in it. Changed the for loop to a while loop... because we pop elements and when there are none left, the while loop will stop.

rayapetkova committed:
Added new group box

New group box, new_layout and two buttons "Favourites" and "Log out"

rayapetkova committed:
Update main_menu

Moved the buttons "Favourites" and "Log out" to the right of the top_buttons_layout.

rayapetkova committed:
Made a few changes in main_menu

Added a basic structure of the creating the categories layout.

karastoyanov committed:
Categories Grid Update

karastoyanov committed:
Categories menu update

karastoyanov committed:
Buttons in GridLayout Updated - almost done

karastoyanov committed:
Merge branch 'development' into main_menu

rayapetkova commented on pyqt5-gui/main_menu.py:
```diff
@@ -120,8 +130,8 @@ def books_open():
+                    category_button.clicked.connect(lambda: functions_dict[current_function_name]())
                     current_vertical_layout.addWidget(category_name)
                     current_vertical_layout.addWidget(category_description)
                     current_vertical_layout.addWidget(category_button)
```
The idea of making the categories groupboxes looks very good. We can add a font in the directory "fonts" and apply it to all the text in the categories_groupbox. Can you show me how can I add another fonts and where can I find them? I will apply it to all the text in categories_groupbox, if you are okay with it.

karastoyanov replied:
For new fonts you can use [dafont.com](https://www.dafont.com/), but there are hundreds of web sites where you can download fonts for free, [Google Fonts](https://fonts.google.com/) is also a good place to take a look.

After you download the font, you need to place it in `fonts` directory, the file is with extention `.otf`, `.ttf`...depending on the font license. Then you can import the font inside the application using the Qt Framework.


```python
"""This is the old JetBrains Font we are using for most of the objects"""
font = QFontDatabase.addApplicationFont(r'/fonts/jetbrains-mono.regular.ttf') 
"""This would be the new font"""
font = QFontDatabase.addApplicationFont(r'/path/to/the/new_font') 
if font >= 0: """<-- Change this to only greather than. If we have 0 custom fonts, it should throw an error"""
   fonts = QFontDatabase.applicationFontFamilies(font)
else:
   print("Error loading fonts!")
```

When setting a custom font to any QLabel() object, you can call the `fonts` array and specify which font exactly you want to use:

```python
some_label = QLabel()
some_label.setText("This is some text label")
some_label.setFont(QFont(fonts[n], 23)) """Where n is the exact fonts array index, in our case 0 or 1"""
```

This should work 🤔

karastoyanov committed:
Calling buttons fixed

Co-authored-by: rayapetkova rayapetkova12321@gmail.com

karastoyanov committed:
Main Menu minor fixes

rayapetkova committed:
Update categories layouts and added new layout with image

- Added three new things in manage_portfolio
- Added a new TEST .jpg file in directory pyqt5-gui
- Added a layout for the test image

rayapetkova commented:
Last changes: 5843bbb
1. db_handle --> manage_portfolio
   - Added a `category_description`, `category_function` and `category_image_URL` variables, which the user should enter
   - Changed the query --> added the three variables in it :point_up_2:


2. pyqt5-gui --> food.jpg
   - Added a random image `food.jpg` for tests


3. pyqt5-gui --> main_menu
   - Made a new label `photo_label` in which we can add images
   - Added the image food.jpg to the photo_label `(photo_label.setPixmap(pixmap))`

rayapetkova committed:
Update import files in login window

Imported only the things we need, and not everything from the libraries

Commit:
Categories Pixmap Fixed

    - image URL taken from database
    - image resized for max 300x100
    - image QLabel object location moved before the button

karastoyanov committed:
Categories Images Stored Locally

karastoyanov committed:
Cleaning unneccesary code

rayapetkova committed:
Update backgrounds

Made an image background for all the groupboxes and set a colour to the labels

karastoyanov committed:
some test commit

karastoyanov committed:
Merge branch 'main_menu' of github.com:Stake-And-Rope/market-app into main_menu

karastoyanov committed:
Categories Colors and Shadows Update

karastoyanov committed:
Small Visual Improvements

rayapetkova committed:
Updated categories' bckgrounds

Added a new test photo with reduced opacity and added it as a background in the category "Accessories"

rayapetkova committed:
Added new images

Found new images, reduced their opacity and added them in img --> categories_new_images

karastoyanov committed:
Code Cleaning

rayapetkova committed:
Deleted old images

- Deleted old images
- Renames the directory of the new images to "categories"
- Changed row 141 --> "categories_new_images" to "categories"
- Row 142: made current groupboxes rounded

rayapetkova committed:
Update categories groupboxes --> category_name label

- Commented the lines that make the shadow
- Added a background for the category_name labels - white color
- Rounded the background of the category_name labels

karastoyanov committed:
Left Buttons Layout minor update

karastoyanov committed:
Minor GUI Updates

karastoyanov committed:
Buttons Styles Update:
 - .setMaximumWidth() value updated to 150
 - custom Font set

karastoyanov committed:
Minor Design Changes
 - Description text removed
 - Objects re-alligned
 - Additional Styling to be implemented in next release

karastoyanov committed:
user info update

karastoyanov committed:
major main_menu GUI rework
- in categories groupbox - current_vertical_layout alignment set to `Bottom`
- favourites button added Top Layout
- left buttons reduced

rayapetkova reviewed (approved):
Ready to be merged
